calendar/engine/user_test.go

```diff
@@ -0,0 +1,700 @@
+package engine
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/config"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/engine/mock_plugin_api"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/engine/mock_welcomer"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/remote/mock_remote"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/store/mock_store"
+	"github.com/mattermost/mattermost-plugin-mscalendar/calendar/utils/bot/mock_bot"
+
+	"github.com/mattermost/mattermost/server/public/model"
+)
+
+func TestExpandUser(t *testing.T) {
+	mscalendar, mockStore, _, _, mockPluginAPI, _, _ := MockSetup(t)
+
+	user := &User{
+		User:             &store.User{Settings: store.Settings{}, Remote: &remote.User{ID: "testRemoteID"}},
+		MattermostUserID: "testMMUserID",
+	}
+
+	tests := []struct {
+		name       string
+		eventID    string
+		setupMock  func()
+		assertions func(t *testing.T, err error)
+	}{
+		{
+			name:    "error expanding remote user",
+			eventID: "testEventID",
+			setupMock: func() {
+				user.User = nil
```

Comment:
I think we should do reverse i.e. Setting the user as nil initially. This will decrease the line of code